### DISCUSSION TITLE
XIAO ESP32S3 Sense Keywords Spotting | Seeed Studio Wiki

### DISCUSSION THREAD

**Discussion by giscus** [Seeed-Studio/wiki-documents#618] [Q&A]
# XIAO ESP32S3 Sense Keywords Spotting | Seeed Studio Wiki

We continue exploring Machine Learning on the giant new tiny device of Seeed XIAO family, the ESP32S3 Sense.

https://wiki.seeedstudio.com/xiao_esp32s3_keyword_spotting/#

**Comment by thedevleon** [comment]
When trying to run the final sketch, I'm getting the following error:

Edge Impulse Inferencing Demo
Inferencing settings:
	Interval: 0.062500 ms.
	Frame size: 16000
	Sample length: 1000 ms.
	No. of classes: 2

Starting continious inference in 2 seconds...
Recording...
AllocateTensors() failedERR: Failed to run classifier (-3)
AllocateTensors() failedERR: Failed to run classifier (-3)
AllocateTensors() failedERR: Failed to run classifier (-3)
AllocateTensors() failedERR: Failed to run classifier (-3)
AllocateTensors() failedERR: Failed to run classifier (-3)
AllocateTensors() failedERR: Failed to run classifier (-3)


Any idea how to fix this?

**Comment by kernaldpark** [comment]
test

**Comment by FyrbyAdditive** [comment]
It's pretty hard to follow this at the time of writing as Edge Impulse seems different and trying to work out missing bits is pretty hard.

**Comment by valtsss** [comment]
I have been dealing with an ongoing error with de version 3.0: error: 'esp_i2s' has not been declared

**Reply by nfs0619** [reply]
Sorry for the inconvenience. Due to version updates, this tutorial may no longer work as expected, and you might need to make some adjustments to use it.

**Reply by maet3608** [reply]
I had the same issue and wrote a tutorial on how to get around the problem:
https://www.makerguides.com/voice-control-with-xiao-esp32-s3-sense-and-edge-impulse/

**Comment by maet3608** [comment]
hmm, maybe useful for all. This tutorial doesn't work anymore but I wrote an updated version that works (for now Aug 2025):
https://www.makerguides.com/voice-control-with-xiao-esp32-s3-sense-and-edge-impulse/

**Reply by nfs0619** [reply]
Thank you for sharing! We truly appreciate your contribution and would love to welcome you to be part of our project.

If you’re interested in learning more about how to get involved or contribute, please check out the following links:

* [Contributor Introduction](https://wiki.seeedstudio.com/contributors/)
* [Contributor Project Page](https://github.com/orgs/Seeed-Studio/projects/6/views/1?pane=issue&itemId=30957479)

We look forward to more communication and collaboration with you!
